src/ShallowTraversal.js

```diff
@@ -108,34 +107,34 @@ export function nodeMatchesObjectProps(node, props) {
   return isSubset(propsOfNode(node), props);
 }
 
+function buildSelectorPredicate(selector) {
+  const nodes = getAst(selector).nodes[0];
+  return AND(nodes.map(node => {
+    switch (node.type) {
+      case 'class':
```

Comment:
Yeah. Just put it in a map so there is a central place of seeing the node types supported